src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -356,11 +360,17 @@ public async Task SaveChangesAsync(IEnumerable<UnspentOutputs> unspentOutputs, I
                 }
 
                 this.blockHash = nextBlockHash;
+                var rewindData = new RewindData(nextBlockHash);
+
                 foreach (UnspentOutputs unspent in unspentOutputs)
                 {
-                    CacheItem existing;
-                    if (this.unspents.TryGetValue(unspent.TransactionId, out existing))
+                    if (this.unspents.TryGetValue(unspent.TransactionId, out CacheItem existing))
                     {
+                        // We'll need to restore the original outputs.
+                        UnspentOutputs clone = unspent.Clone();
+                        clone.Outputs = existing.UnspentOutputs?.Outputs.ToArray() ?? Array.Empty<TxOut>();
```

Comment:
So do I understand that correctly –  `existing` is the old version before the new block was applied and so we want to have the old version of outputs array in the rewind data?

Reply:
Yes, this is my understanding as well. We want to get original outputs before we mutate it.